package/tests/test_api/test_graphql/test_mutations.py

```diff
@@ -0,0 +1,321 @@
+        self, client, data_access_manager_with_no_run
+    ):
+        # add a new run
+        example_run_id = "test_id"
+        run = RunModel(
+            id=example_run_id,
+            blob=json.dumps(
+                {"session_id": example_run_id, "cli": {"command_path": "kedro run"}}
+            ),
+        )
+        data_access_manager_with_no_run.runs.add_run(run)
+
+        query = f"""
+            mutation updateRun {{
+              updateRunDetails(runId: "{example_run_id}", runInput: {{bookmark: true}}) {{
+                __typename
+                ... on UpdateRunDetailsSuccess {{
+                    run {{
+                        id
+                        title
+                        bookmark
+                        notes
+                    }}
+                }}
+                ... on UpdateRunDetailsFailure {{
+                    id
+                    errorMessage
+                }}
+              }}
+            }}
+        """
+
+        response = client.post("/graphql", json={"query": query})
+        assert response.json() == {
+            "data": {
+                "updateRunDetails": {
+                    "__typename": "UpdateRunDetailsSuccess",
+                    "run": {
+                        "id": example_run_id,
+                        "bookmark": True,
+                        "title": example_run_id,
+                        "notes": "",
+                    },
+                }
+            }
+        }
+
+    def test_update_user_details_should_update_when_details_exist(
+        self, client, example_run_ids
+    ):
+        example_run_id = example_run_ids[0]
+        query = f"""
+            mutation updateRun {{
+              updateRunDetails(runId: "{example_run_id}", runInput: {{title:"new title", notes: "new notes"}}) {{
+                __typename
+                ... on UpdateRunDetailsSuccess {{
+                    run {{
+                        id
+                        title
+                        bookmark
+                        notes
+                    }}
+                }}
+                ... on UpdateRunDetailsFailure {{
+                    id
+                    errorMessage
+                }}
+              }}
+            }}
+        """
+
+        response = client.post("/graphql", json={"query": query})
+        assert response.json() == {
+            "data": {
+                "updateRunDetails": {
+                    "__typename": "UpdateRunDetailsSuccess",
+                    "run": {
+                        "id": example_run_id,
+                        "bookmark": False,
+                        "title": "new title",
+                        "notes": "new notes",
+                    },
+                }
+            }
+        }
+
+    def test_update_user_details_should_fail_when_run_doesnt_exist(self, client):
+        response = client.post(
+            "/graphql",
+            json={
```

Comment:
it's a lot more readable in JS land. We can change f-string to `%s` or `.format` maybe... I asked @rashidakanchwala to use f-string but in retrospect I think it was a mistake.

Reply:
To be honest I'm not sure `%s` or `.format` will make a huge difference to readability compared to f-strings. It would be nice if there was a completely different way of doing it somehow, but no worries at all if not - the f-strings are ok.